apps/omg_watcher_info/lib/omg_watcher_info/transaction.ex

```diff
@@ -0,0 +1,195 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.Transaction do
+  @moduledoc """
+  Module create transaction from selected utxos and order.
+  """
+
+  alias OMG.Crypto
+  alias OMG.State.Transaction
+  alias OMG.TypedDataHash
+  alias OMG.WatcherInfo.DB
+  alias OMG.WatcherInfo.UtxoSelection
+
+  require Transaction.Payment
+
+  @empty_metadata <<0::256>>
+
+  @type create_t() ::
+          {:ok, nonempty_list(transaction_t())}
+          | {:error, :too_many_inputs}
+          | {:error, :too_many_outputs}
+          | {:error, :empty_transaction}
+
+  @type transaction_t() :: %{
+          inputs: nonempty_list(%DB.TxOutput{}),
+          outputs: nonempty_list(UtxoSelection.payment_t()),
+          fee: UtxoSelection.fee_t(),
+          txbytes: Transaction.tx_bytes() | nil,
+          metadata: Transaction.metadata(),
+          sign_hash: Crypto.hash_t() | nil,
+          typed_data: TypedDataHash.Types.typedDataSignRequest_t()
+        }
+
+  @type order_t() :: %{
+          owner: Crypto.address_t(),
+          payments: nonempty_list(UtxoSelection.payment_t()),
+          metadata: binary() | nil,
+          fee: UtxoSelection.fee_t()
+        }
+
+  @type utxos_map_t() :: %{UtxoSelection.currency_t() => UtxoSelection.utxo_list_t()}
```

Comment:
Can we just use the type defined in `utxo_selection.ex` instead of redefining it ?